app/rpc/namespaces/eth/api.go

```diff
@@ -946,11 +967,18 @@ func (api *PublicEthereumAPI) doCall(
 	if err != nil {
 		return nil, err
 	}
-
+	simulateQueryData := sdk.QuerySimulateData{
```

Comment:
how about rename QuerySimulateData to TxStateAccessor or some else? QuerySimulateData seems too general. we can refer this doc: https://geth.ethereum.org/docs/rpc/ns-eth

Reply:
this QuerySimulateData struct is used to pass the params from any caller to query the ABCI path "app/simulate". not only for this case.  Maybe it's better change to SimulateQueryData{}